[native-image] java.net.NetworkInterface.isUp0 not implemented

As in PR #1076 I am also getting an UnsatisfiedLinkError when I try to run my native executable. Although the required underlying methods (getFlags0 and openSocketWithFallBack()) have been implemented as part of PR #1406, it seems like isUp() is still missing. Are there plans to add isUp()?

Example stack trace:

```
Exception in thread "main" java.util.ServiceConfigurationError: liquibase.sqlgenerator.SqlGenerator: Provider liquibase.sqlgenerator.core.LockDatabaseChangeLogGenerator could not be instantiated
        at java.util.ServiceLoader.fail(ServiceLoader.java:232)
        at java.util.ServiceLoader.access$100(ServiceLoader.java:185)
        at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:384)
        at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
        at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
    	...
Caused by: java.lang.UnsatisfiedLinkError: java.net.NetworkInterface.isUp0(Ljava/lang/String;I)Z [symbol: Java_java_net_NetworkInterface_isUp0 or Java_java_net_NetworkInterface_isUp0__Ljava_lang_String_2I]
        at com.oracle.svm.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:145)
        at com.oracle.svm.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:57)
        at java.net.NetworkInterface.isUp0(NetworkInterface.java)
        at java.net.NetworkInterface.isUp(NetworkInterface.java:399)
        at liquibase.util.NetUtil.getLocalHost(NetUtil.java:31)
        at liquibase.util.NetUtil.getLocalHostName(NetUtil.java:69)
        at liquibase.sqlgenerator.core.LockDatabaseChangeLogGenerator.<clinit>(LockDatabaseChangeLogGenerator.java:30)
        at com.oracle.svm.core.hub.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:350)
        at com.oracle.svm.core.hub.ClassInitializationInfo.initialize(ClassInitializationInfo.java:270)
        at java.lang.Class.ensureInitialized(DynamicHub.java:451)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.lang.Class.newInstance(DynamicHub.java:762)
        at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:380)
        ...
```

--- a/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java
+++ b/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java
@@ -4792,6 +4792,21 @@ final class Target_java_net_NetworkInterface {
         return ((ret & NetIf.IFF_LOOPBACK()) != 0) ? true : false;
     }
 
+
+    // 463   /*
+    // 464    * Class:     java_net_NetworkInterface
+    // 465    * Method:    isUp0
+    // 466    * Signature: (Ljava/lang/String;I)Z
+    // 467    */
+    @Substitute
+    @SuppressWarnings({"unused"})
+    private static boolean isUp0(String name, int ind) throws SocketException {
+        //    471    int ret = getFlags0(env, name);
+        int ret = JavaNetNetworkInterface.getFlags0(name);
+        //    472    return ((ret & IFF_UP) && (ret & IFF_RUNNING)) ? JNI_TRUE :  JNI_FALSE;
+        return (((ret & NetIf.IFF_UP()) != 0) && (ret & NetIf.IFF_RUNNING()) != 0);
+    }
+
     /*
      * Translated from jdk8u/jdk/src/solaris/native/java/net/NetworkInterface.c
      */

--- a/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/NetIf.java
+++ b/substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/headers/NetIf.java
@@ -46,6 +46,12 @@ import org.graalvm.word.PointerBase;
 public class NetIf {
 
     @CConstant
+    public static native int IFF_UP();
+
+    @CConstant
+    public static native int IFF_RUNNING();
+
+    @CConstant
     public static native int IFF_LOOPBACK();
 
     // #define IF_NAMESIZE 16

--- a/substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/NetworkInterfaceTest.java
+++ b/substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/NetworkInterfaceTest.java
@@ -44,4 +44,16 @@ public class NetworkInterfaceTest {
 
         Assert.assertTrue("At least one loopback found", foundLoopback);
     }
+
+    @Test
+    public void testIsUp() throws SocketException {
+        Enumeration<NetworkInterface> ifaces = NetworkInterface.getNetworkInterfaces();
+        boolean somethingIsUp = false;
+        while (ifaces.hasMoreElements()) {
+            NetworkInterface each = ifaces.nextElement();
+            somethingIsUp = somethingIsUp || each.isUp();
+        }
+        Assert.assertTrue("At least one interface is up", somethingIsUp);
+    }
+
 }

